### PR TITLE
Add symlink support to JGit

### LINKED ISSUE #636: [BUG] symlinks are clobbered by the app
**Describe the bug**
A password will fail to decrypt if it is a soft link as opposed to a normal file.

**To Reproduce**
1. Create a password that's a soft link to an existing password using `ln -s <passwd>.gpg softlink.gpg`
2. Pull the repository on device
3. Try to decrypt the password

**Expected behavior**
Password is decrypted and shown in the UI

**Device information (please complete the following information):**
 - Device: Google Pixel 2
 - OS: Android R DP1
 - App version 1.6.0-SNAPSHOT

**Additional context**

Relevant log snippet

```
E  onError getErrorId: 0
E  onError getMessage: No valid OpenPGP encrypted or signed data found!
```

### PR DESCRIPTION
**Pull request by fmeum**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description

Enable symlink support in JGit.

Symlink support is provided by the `FS_POSIX_Java7` file system, which is loaded dynamically from the `org.eclipse.jgit.java7` module. Previously, we didn't include that module, which meant that symlinks were treated as ordinary files as the relevant check [here](https://github.com/eclipse/jgit/blob/2383cccf1226ad4c3140846517458de1e5283c27/org.eclipse.jgit/src/org/eclipse/jgit/dircache/DirCacheCheckout.java#L1228) would fail for the `FS_POSIX_Java6` filesystem provider.

This probably requires users to delete and re-checkout their repositories to get their symlinks to work. @msfjarvis I'm unsure about how to communicate this best, would a changelog entry be enough?

## :bulb: Motivation and Context

Fixes #636.

## :green_heart: How did you test it?
Cloned the https://github.com/android-password-store/pass-test repository and observed that the symlinked file is decrypted correctly.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -117,6 +118,8 @@ dependencies {
     implementation(Dependencies.ThirdParty.jgit) {
         exclude(group = "org.apache.httpcomponents", module = "httpclient")
     }
+    // Loaded dynamically by JGit to provider symlink support
```
Comment (fmeum) [inline app/build.gradle.kts]:
```suggestion
    // Loaded dynamically by JGit to provide symlink support
```

**Comment by msfjarvis** [comment]
Can you check if release builds need a R8 rule?

**Comment by fmeum** [comment]
This is actually precisely the fix @Dioxo found [here](https://github.com/android-password-store/Android-Password-Store/issues/636#issuecomment-637750154), not sure how I missed that back then.

**Comment by fmeum** [comment]
> Can you check if release builds need a R8 rule?

Not sure why, but it does not seem to require a rule.

**Comment by msfjarvis** [comment]
> This is actually precisely the fix @Dioxo found [here](https://github.com/android-password-store/Android-Password-Store/issues/636#issuecomment-637750154), not sure how I missed that back then.

It didn't work back when I tried it which was why it never landed. Guess the problem was that we didn't think of existing files being clobbered completely and didn't try with fresh clones.

> > Can you check if release builds need a R8 rule?
> 
> Not sure why, but it does not seem to require a rule.

Interesting...

**Comment by msfjarvis** [comment]
How did you create the symlink? Using `ln -s ../domain1.com/username.gpg .` resulted in a password I couldn't decrypt.

**Comment by fmeum** [comment]
> How did you create the symlink? Using `ln -s ../domain1.com/username.gpg .` resulted in a password I couldn't decrypt.

I just did `ln -s test_with_email.gpg symlink.gpg` in the root our `pass-test` repo. Maybe the relative path messes this up?

**Comment by msfjarvis** [comment]
> > How did you create the symlink? Using `ln -s ../domain1.com/username.gpg .` resulted in a password I couldn't decrypt.
> 
> I just did `ln -s test_with_email.gpg symlink.gpg` in the root our `pass-test` repo. Maybe the relative path messes this up?

It certainly does. Doing `ln -s domain1.com domain2.com` fixed the problem. I don't think a changelog entry is going to cut it if we have all these edge cases to worry about.

**Comment by fmeum** [comment]
> > > Can you check if release builds need a R8 rule?
> > 
> > 
> > Not sure why, but it does not seem to require a rule.
> 
> Interesting...

Learned about `-whyareyoukeeping`, which provides the reason:
```
org.eclipse.jgit.util.FS_POSIX_Java7
|- is instantiated in:
|  org.eclipse.jgit.util.FS org.eclipse.jgit.util.Java7FSFactory.detect(java.lang.Boolean)
|- is overriding method:
|  org.eclipse.jgit.util.FS org.eclipse.jgit.util.FS$FSFactory.detect(java.lang.Boolean)
|- is invoked from:
|  org.eclipse.jgit.util.FS org.eclipse.jgit.util.FS.detect()
|- is invoked from:
|  void org.eclipse.jgit.util.FS.<clinit>()
|- is reachable from:
|  org.eclipse.jgit.util.FS
|- is referenced from:
|  org.eclipse.jgit.dircache.DirCache org.eclipse.jgit.lib.Repository.readDirCache()
|- is invoked from:
|  org.eclipse.jgit.lib.RepositoryState org.eclipse.jgit.lib.Repository.getRepositoryState()
|- is invoked from:
|  void com.zeapo.pwdstore.git.GitConfigActivity.onCreate(android.os.Bundle)
|- is defined in library method overridden by:
|  com.zeapo.pwdstore.git.GitConfigActivity
```

**Comment by fmeum** [comment]
> > > How did you create the symlink? Using `ln -s ../domain1.com/username.gpg .` resulted in a password I couldn't decrypt.
> > 
> > 
> > I just did `ln -s test_with_email.gpg symlink.gpg` in the root our `pass-test` repo. Maybe the relative path messes this up?
> 
> It certainly does. Doing `ln -s domain1.com domain2.com` fixed the problem. I don't think a changelog entry is going to cut it if we have all these edge cases to worry about.

Maybe we can fix this on our end by sprinkling some more `absolutePath`s in the relevant part of the codebase? Not sure though, symlinks are weird.

**Comment by msfjarvis** [comment]
> > > > How did you create the symlink? Using `ln -s ../domain1.com/username.gpg .` resulted in a password I couldn't decrypt.
> > > 
> > > 
> > > I just did `ln -s test_with_email.gpg symlink.gpg` in the root our `pass-test` repo. Maybe the relative path messes this up?
> > 
> > 
> > It certainly does. Doing `ln -s domain1.com domain2.com` fixed the problem. I don't think a changelog entry is going to cut it if we have all these edge cases to worry about.
> 
> Maybe we can fix this on our end by sprinkling some more `absolutePath`s in the relevant part of the codebase? Not sure though, symlinks are weird.

I'll look into it.

**Comment by fmeum** [comment]
I just created what I believe to be a relative symlink and decryption works fine:
```
~/git/pass-test/relative_symlink$ ln -s ../subdir_with_fingerprint/test_with_fingerprint.gpg .
```

**Comment by msfjarvis** [comment]
> I just created what I believe to be a relative symlink and decryption works fine:
> 
> ```
> ~/git/pass-test/relative_symlink$ ln -s ../subdir_with_fingerprint/test_with_fingerprint.gpg .
> ```

:thinking: not sure what's up on my end then.

**Review by msfjarvis — APPROVED** [review]
